File: .config/hiss/testdata/HISS-02/go/negative/bounded-for.go
// SPDX-FileCopyrightText: 2026 lusoris <lusoris@pm.me>
//
// SPDX-License-Identifier: EUPL-1.2

// Package p holds the bounded equivalent of the positive fixture.
package p

// maxDrain is the scalar upper bound HISS-02 requires.
const maxDrain = 1024

// Drain reads at most maxDrain values, so the loop terminates.
func Drain(ch <-chan int) int {
	total := 0
	for i := 0; i < maxDrain; i++ {
		total += <-ch
	}
	return total
}

File: .config/hiss/testdata/HISS-02/c/negative/bounded-for.c
// SPDX-FileCopyrightText: 2026 lusoris <lusoris@pm.me>
//
// SPDX-License-Identifier: EUPL-1.2

#define MAX_POLLS 1024

int poll_bounded(volatile int *flag)
{
	for (int i = 0; i < MAX_POLLS; i++) {
		if (*flag)
			return 1;
	}
	return 0;
}
